[Bug]:

start error ， maybe jar import error
you can use my demo
error:
Exception in thread "main" java.lang.NoClassDefFoundError: com/fasterxml/jackson/datatype/jsr310/JavaTimeModule
	at io.agentscope.core.util.JsonUtils.<clinit>(JsonUtils.java:46)
	at io.agentscope.core.model.OpenAIClient.call(OpenAIClient.java:275)
	at io.agentscope.core.model.OpenAIChatModel.lambda$doStream0$1(OpenAIChatModel.java:156)
	at reactor.core.publisher.FluxDefer.subscribe(FluxDefer.java:47)
	at reactor.core.publisher.InternalFluxOperator.subscribe(InternalFluxOperator.java:67)
	at reactor.core.publisher.FluxRetryWhen.subscribe(FluxRetryWhen.java:81)
	at reactor.core.publisher.FluxRetryWhen.subscribeOrReturn(FluxRetryWhen.java:87)
	at reactor.core.publisher.Flux.subscribe(Flux.java:8873)
	at reactor.core.publisher.MonoFlatMapMany$FlatMapManyMain.onNext(MonoFlatMapMany.java:198)
	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.complete(MonoIgnoreThen.java:297)
	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.onNext(MonoIgnoreThen.java:191)
	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.subscribeNext(MonoIgnoreThen.java:240)
	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.onComplete(MonoIgnoreThen.java:207)
	at reactor.core.publisher.Operators.complete(Operators.java:137)
	at reactor.core.publisher.MonoEmpty.subscribe(MonoEmpty.java:46)
	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:54)
	at reactor.core.publisher.Mono.subscribe(Mono.java:4569)
	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.subscribeNext(MonoIgnoreThen.java:268)
	at reactor.core.publisher.MonoIgnoreThen.subscribe(MonoIgnoreThen.java:51)
	at reactor.core.publisher.Flux.subscribe(Flux.java:8888)
	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.subscribeNext(MonoIgnoreThen.java:268)
	at reactor.core.publisher.MonoIgnoreThen.subscribe(MonoIgnoreThen.java:51)
	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:75)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:166)
	at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onNext(FluxMapFuseable.java:130)
	at reactor.core.publisher.Operators$ScalarSubscription.request(Operators.java:2565)
	at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.request(FluxMapFuseable.java:172)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.request(MonoFlatMap.java:195)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.request(MonoFlatMap.java:195)
	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.set(Operators.java:2361)
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onSubscribe(FluxOnErrorResume.java:75)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onSubscribe(MonoFlatMap.java:118)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onSubscribe(MonoFlatMap.java:118)
	at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onSubscribe(FluxMapFuseable.java:96)
	at reactor.core.publisher.MonoJust.subscribe(MonoJust.java:56)
	at reactor.core.publisher.Mono.subscribe(Mono.java:4569)
	at reactor.core.publisher.Mono.block(Mono.java:1772)
	at AA.main(AA.java:79)
Caused by: java.lang.ClassNotFoundException: com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	... 38 more


demo:

import io.agentscope.core.ReActAgent;
import io.agentscope.core.message.Msg;
import io.agentscope.core.model.OpenAIChatModel;
import io.agentscope.core.model.transport.HttpTransportConfig;
import io.agentscope.core.model.transport.JdkHttpTransport;
import io.agentscope.core.tool.Tool;
import io.agentscope.core.tool.ToolParam;
import io.agentscope.core.tool.Toolkit;
import io.agentscope.core.tool.mcp.McpSyncClientWrapper;
import io.modelcontextprotocol.client.McpClient;
import io.modelcontextprotocol.client.McpSyncClient;
import io.modelcontextprotocol.client.transport.HttpClientSseClientTransport;
import reactor.core.publisher.Mono;

import javax.net.ssl.SSLContext;
import javax.net.ssl.SSLParameters;
import java.net.http.HttpClient;
import java.util.Set;

public class AA {

    @Tool(description = "根据配置的名字获取id")
    public String findConfigIdByName(@ToolParam(name = "配置类型", description = "1:链路,2:应用") Integer type, String name) {
        System.out.println(type + " : " + name);
        if(type == 1){
            return name.equals("abc链路") ? "a12345" : "yuoid1234";
        }
        return name.equals("http错误包") ? "30789" : "30001";

    }

    public static void main(String[] args) throws Exception {
        ReActAgent.Builder builder = ReActAgent.builder();
        SSLParameters sslParameters = new SSLParameters();
        sslParameters.setWantClientAuth(false);
        sslParameters.setUseCipherSuitesOrder(false);
        sslParameters.setEndpointIdentificationAlgorithm(""); // 空字符串 = disable
        sslParameters.setNeedClientAuth(false);
        HttpClient client2 = HttpClient.newBuilder()
                .sslParameters(sslParameters).build();

        builder.model(OpenAIChatModel.builder()
                .httpTransport(new JdkHttpTransport(client2,HttpTransportConfig.defaults()))
                .modelName("Qwen3-Coder-30B-A3B-Instruct").apiKey("11111")
                .baseUrl("https://xxx.xx.xxx").build());


        McpClient.SyncSpec sync = McpClient.sync(HttpClientSseClientTransport.builder("http://127.0.0.1:8080/")
                .sseEndpoint("/sse")
                .customizeClient(o -> o.sslParameters(sslParameters))
                .build());

        McpSyncClient client = sync.build();
        Toolkit toolkit = new Toolkit();
        toolkit.registerMcpClient(new McpSyncClientWrapper("test",client));
        builder.toolkit(toolkit);
        Set<String> toolNames = toolkit.getToolNames();
        System.out.println(toolNames);
        builder.maxIters(100);
        builder.name("测试agent");


        String jsonTemplate = """
                {"probeId":"链路id","filter":"过滤条件","date":"时间范围","table":"表"}
                """;
        String question = "分析 链路：abc链路，最近10分钟，表：应用，过滤条件:http错误包";
        String desc = String.format("""
                你是个 智能查询转换 的专家,需要多次调用tool来获取结果；当前时间是：2026-01-10 19:32
                1.需要调用工具查询到名字对应的id，没有的key，你可以直接填空字符
                2.如果是过滤条件的话，是与或关系表达式
                表：是应用的话，输出application_id=应用的id
                最终用json返回:%s
                3.时间的输出是:20xx-xx-xx xx:xx:xx - 20xx-xx-xx xx:xx:xx 从小到大
                """, jsonTemplate);
        builder.sysPrompt(desc);
        ReActAgent agent = builder.build();

        Mono<Msg> who = agent.call(Msg.builder().name("消息").textContent(question).build());
        Msg block = who.block();
        System.out.println(block.getTextContent());
    }
}

--- a/agentscope-distribution/agentscope-all/pom.xml
+++ b/agentscope-distribution/agentscope-all/pom.xml
@@ -218,6 +218,12 @@
             <groupId>com.fasterxml.jackson.core</groupId>
             <artifactId>jackson-annotations</artifactId>
         </dependency>
+
+        <!-- Jackson Java 8 Date/Time support (LocalDateTime, Instant, etc.) -->
+        <dependency>
+            <groupId>com.fasterxml.jackson.datatype</groupId>
+            <artifactId>jackson-datatype-jsr310</artifactId>
+        </dependency>
     </dependencies>
 
     <build>

